### PR TITLE
feat: add LiquidityPoolService; add LP deposit(), withdraw(), claimRewards() method;

### PR DESCRIPTION
**Pull request by Kozer4**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -440,4 +459,86 @@ export class AllbridgeCoreSdk {
...
+  ): Promise<UserBalanceInfo> {
+    return this.liquidityPoolService.getUserBalanceInfo(
+      accountAddress,
+      token,
+      provider
+    );
+  }
+
+  /**
+   * Calculates the lp amount of tokens will be deposited
```
Comment (m-cote) [inline src/index.ts:483]:
```suggestion
   * Calculates the amount of LP tokens that will be deposited
```

File: examples/liquidity-pool/trx/get-lp-amount-on-deposit.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  const oneToken = 1;
+  const estimatedAmount = await sdk.getLPAmountOnDeposit(
+    oneToken,
+    tokenInfo,
+    tronWeb
+  );
+
+  console.log(
+    "If you send %d , then %d will be deposited",
```
Comment (m-cote) [inline examples/liquidity-pool/trx/get-lp-amount-on-deposit.js:30]:
```suggestion
    "If you send %d , then %d of LP tokens will be deposited",
```

File: documentation/core-sdk-liquidity-pools-api.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+_Returns_:
+
+* poolInfo: PoolInfo
+
+### Calculate LP amount will be deposited
+
+_Method_: getLPAmountOnDeposit
+
+Calculates the lp amount of tokens will be deposited
```
Comment (m-cote) [inline documentation/core-sdk-liquidity-pools-api.md:108]:
```suggestion
Calculates the amount of LP tokens that will be deposited
```

File: examples/liquidity-pool/evm/get-lp-amount-on-deposit.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  );
+
+  const oneToken = 1;
+  const estimatedAmount = await sdk.getLPAmountOnDeposit(
+    oneToken,
+    tokenInfo,
+    web3
+  );
+  console.log(
+    "If you send %d , then %d will be deposited",
```
Comment (m-cote) [inline examples/liquidity-pool/evm/get-lp-amount-on-deposit.js:26]:
```suggestion
    "If you send %d , then %d LP tokens will be deposited",
```

File: examples/liquidity-pool/solana/get-lp-amount-on-deposit.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  const sdk = new AllbridgeCoreSdk(development);
+  const tokenInfo = (await sdk.tokens()).find(
+    (tokenInfo) => tokenInfo.tokenAddress === tokenAddress
+  );
+
+  const oneToken = 1;
+  const estimatedAmount = await sdk.getLPAmountOnDeposit(oneToken, tokenInfo);
+
+  console.log(
+    "If you send %d , then %d will be deposited",
```
Comment (m-cote) [inline examples/liquidity-pool/solana/get-lp-amount-on-deposit.js:16]:
```suggestion
    "If you send %d , then %d LP tokens will be deposited",
```

File: examples/liquidity-pool/evm/get-amount-to-be-withdrawn.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  const halfToken = 0.5;
+  const estimatedAmount = await sdk.getAmountToBeWithdrawn(
+    halfToken,
+    accountAddress,
+    tokenInfo,
+    web3
+  );
+
+  console.log(
+    "If you request %d , then %d will be received",
```
Comment (m-cote) [inline examples/liquidity-pool/evm/get-amount-to-be-withdrawn.js:29]:
```suggestion
    "If you withdraw %d LP tokens, then %d will be received",
```

File: examples/liquidity-pool/solana/get-amount-to-be-withdrawn.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+  const halfToken = 0.5;
+  const estimatedAmount = await sdk.getAmountToBeWithdrawn(
+    halfToken,
+    accountAddress,
+    tokenInfo
+  );
+
+  console.log(
+    "If you request %d , then %d will be received",
```
Comment (m-cote) [inline examples/liquidity-pool/solana/get-amount-to-be-withdrawn.js:21]:
```suggestion
    "If you withdraw %d LP tokens, then %d will be received",
```

File: examples/liquidity-pool/trx/get-amount-to-be-withdrawn.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  const halfToken = 0.5;
+  const estimatedAmount = await sdk.getAmountToBeWithdrawn(
+    halfToken,
+    accountAddress,
+    tokenInfo,
+    tronWeb
+  );
+
+  console.log(
+    "If you request %d , then %d will be received",
```
Comment (m-cote) [inline examples/liquidity-pool/trx/get-amount-to-be-withdrawn.js:32]:
```suggestion
    "If you withdraw %d LP tokens, then %d will be received",
```

**Review by m-cote — COMMENTED** [review]
We can make it more clear in the docs and examples that we deposit tokens and get LP tokens in return, and after that we can withdraw LP tokens to get tokens.

File: examples/get-tokens-info.js [hunk 9]
```diff
@@ -2,6 +2,7 @@ const { AllbridgeCoreSdk } = require("@allbridge/bridge-core-sdk");
 
 async function runExample() {
   const sdk = new AllbridgeCoreSdk();
+  // const sdk = new AllbridgeCoreSdk();
```
Comment (m-cote) [inline examples/get-tokens-info.js:5]: ??
```suggestion
```

**Review by m-cote — APPROVED** [review]

File: examples/get-tokens-info.js (hunk 9 above)
Comment (Kozer4) [reply examples/get-tokens-info.js]: thx, fixed

**Pull request merged**